src/core/js/models/lockingModel.js

```diff
@@ -0,0 +1,143 @@
+define([
+	'backbone'
+], function() {
+
+	var set = Backbone.Model.prototype.set;
+
+	_.extend(Backbone.Model.prototype, {
+
+		set: function(attrName, attrVal, options) {
+			var stopProcessing = !this.lockedAttributes || typeof attrName === "object" || typeof attrVal !== "boolean" || !this.isLocking(attrName);
+			if (stopProcessing) return set.apply(this, arguments);
+
+			var isSettingValueForSpecificPlugin = options && options.pluginName;
+			if (!isSettingValueForSpecificPlugin) {
+				console.error("Must supply a pluginName to change a locked attribute");
+				options.pluginName = "compatibility";
```

Comment:
console.error is fine, it was the following line (options.pluginName = "compatibility";) that causes the error
"Uncaught TypeError: Cannot set property 'pluginName' of undefined" with the plugins described. 

I'm being overly picky but it was more the awareness of a break in build that is caused with certain plugin version combinations that I was interested in. Just need to add a check for options before defining pluginName = compatibility, should then work with the v2.0 plugins that set the locked attribute :)